BaseListener/UseCase/DirectDebitProcessUseCase.cs

```diff
@@ -0,0 +1,38 @@
+using BaseListener.Gateway.Interfaces;
+using System.Threading.Tasks;
+using BaseListener.UseCase.Interfaces;
+using Amazon.Lambda.APIGatewayEvents;
+using System.Text.Json;
+using BaseListener.Boundary.Response;
+using BaseListener.Helpers.GeneralModels;
+using BaseListener.Factories;
+using System.Net;
+
+namespace BaseListener.UseCase
+{
+    public class DirectDebitProcessUseCase : IDirectDebitProcessUseCase
+    {
+        private readonly IHttpApiGateway _gateway;
+
+        public DirectDebitProcessUseCase(IHttpApiGateway gateway)
+        {
+            _gateway = gateway;
+        }
+
+        public async Task<APIGatewayProxyResponse> ProcessExecuteAsync(APIGatewayProxyRequest apiGatewayProxyRequest)
+        {
+            APIGatewayProxyResponse response = await _gateway.GetAsync(apiGatewayProxyRequest).ConfigureAwait(false);
+
+            if (response.StatusCode == (int) HttpStatusCode.OK)
+            {
+                PaginatedResponse<TransactionResponse> model = JsonSerializer.Deserialize<PaginatedResponse<TransactionResponse>>(response.Body);
+
+                apiGatewayProxyRequest.Body = JsonSerializer.Serialize(model.Results.ToCalculateAmount());
```

Comment:
The use case shouldn't be dealing with the data access implementation logic (it should be business rules only).

What this means is that we shouldn't be accessing data via `.Body` properties &  deserializing it. The Use Case should receive a concrete Domain layer model from its data access call to the data access layer.

Reply:
Same on Line 28 and maybe line 32

Reply:
Yeah, if you just create the TransactionDomain model, then your Data Access call on Line 28 should be returning an IEnumerable<TransactionDomain> _(that is if you are not going to use the pagination details information)_

Reply:
Done